src/chrome/chromeDebugSession.ts

```diff
@@ -124,8 +124,9 @@ export class ChromeDebugSession extends LoggingDebugSession implements IObservab
      * Overload dispatchRequest to the debug adapters' Promise-based methods instead of DebugSession's callback-based methods
      */
     protected dispatchRequest(request: DebugProtocol.Request): void {
+        const telemetryPropertyCollector = new TelemetryPropertyCollector();
```

Comment:
What are the advantages of creating the TelemetryPropertyCollector here instead of receiving it from the reportTelemetry method?

Reply:
So less methods need to know the contract of requestHandler methods.

Reply:
I think they are equivalent.

Reply:
What do you think would make the code simpler and easier to maintain?
A. Keeping as much of the telemetry implementation inside the reportTelemetry method, and having reportTelemetry create the telemetryPropertyCollector so that the business methods don't need to deal with details of the telemetry implementation?
B. Spreading the telemetry implementation between reportTelemetry and the business methods, and having each business method that uses telemetry having to create it's own telemetryPropertyCollector?